src/RenderCapableBlockTrait.php

```diff
@@ -2,23 +2,44 @@
 
 namespace Dhii\Output;
 
+use Dhii\Output\Exception\TemplateRenderExceptionInterface;
 use Exception as RootException;
 use Dhii\Output\Exception\CouldNotRenderExceptionInterface;
 use Dhii\Output\Exception\RendererExceptionInterface;
 use Psr\Container\ContainerInterface;
 use Dhii\Util\String\StringableInterface as Stringable;
 
 /**
- * Common functionality for template rendering.
+ * Functionality for rendering an internal template with internal context.
  *
- * @since 0.1
+ * @since [*next-version*]
  */
-abstract class AbstractTemplateBlock extends AbstractBlock
+trait RenderCapableBlockTrait
```

Comment:
I still feel like the name does not fully reflect what this trait is, which is _"Functionality for rendering an internal template with internal context"_. Having the name only mention "block" and the description only mention "templates" creates discrepancy and confusion IMO.

Before I approve/reject, I would like to hear your reasoning for choosing this name and this description.

Reply:
There's a `render()` method, so the trait name starts with `RenderCapable`. Like you mentioned before, it's indended for blocks, which contain both the context, and some template definition.

Reply:
I see. I would expect then that the name would reflect the use of templates internally, since I can apply the same thinking process to take a trait that has `_render()` and contains an internal render string and name it the same thing, i.e. `RenderCapableBlockTrait`. Because it is capable of rendering and is intended for blocks.

Reply:
That's what it's called: `RenderCapableBlockTrait`. Where's the conflict?

Reply:
I feel like the name is too generic and applies to any rendering mechanism a block may have. In this case, it is a template mechanism.

As I just explained, a block that simply renders an internal string can also be named `RenderCapableBlockTrait` if your reasoning is applied. As such, I feel like the name of such traits should reflect how the rendering is achieved to prevent future conflicts.

Let's say we decide to create the hypothetical trait I mentioned above. We can't name it `RenderCapableBlockTrait`. It would have to be named `StringRenderCapableBlockTrait` or something.

Thus in anticipation of that, and to also be conceptually correct, I would like to see this trait renamed to something that reflects the use of templates, such as `TemplateRenderCapableBlockTrait`.

Reply:
`TemplateRenderCapableBlockTrait` means there's a `_templateRender()` method.

Reply:
You didn't comment on anything I said, except for the name suggestion that I made simply to make the point I was trying to make clearer.

That said, I'm perfectly fine with bouncing back and forth until we reach some form of conclusion. Suggestion no.2: `RenderCapableTemplateBlockTrait`.

Reply:
I understood the point. I just rejected the name. `RenderCapableTemplateBlockTrait` sounds fine, thanks.

Reply:
This is what an adult discussion looks like, kids. 👨‍💼

Reply:
Time is of the essence.